FinancialTracker/ViewControllers/TabBarController.swift

```diff
@@ -45,6 +62,26 @@ class TabBarController: UITabBarController {
                 default:
                     self.present(UIAlertController.create(title: "Unknown Error", message: "Unknown"), animated: true)
                 }
+            } else {
+                guard let user = user, let balance = user.balance else {
+                    return
+                }
+
+                if !user.premium {
+                    return
+                }
+                
+                if user.expenses.isEmpty, user.incomes.isEmpty {
+                    return
+                }
+                
+                if balance < 100 {
```

Comment:
These random magic numbers for the balance should go into constants, or even better - an enum of some kind.